observ/watcher.py

```diff
@@ -14,14 +14,20 @@
 
 
 def traverse(obj):
+    """
+    Recursively traverse the whole tree to make sure
+    that all values have been 'get'
+    """
     _traverse(obj, set())
 
 
-def _traverse(obj, seen):
+def _traverse(obj, seen: set):
     seen.add(id(obj))
-    if isinstance(obj, dict):
+    if isinstance(obj, Mapping):
```

Comment:
Changing `dict` to `Mapping` and `(list, tuple, set)` to `Container` was key in making the nested array reactive.

Reply:
I added a test to show you why you still need to do that. See `test_dict_iter`

Reply:
Good point!